subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/rich/RichConsoleBasicGroupedTaskLoggingFunctionalTest.groovy

```diff
@@ -48,6 +48,22 @@ class RichConsoleBasicGroupedTaskLoggingFunctionalTest extends AbstractBasicGrou
         when:
         fails('failing')
 
+        then:
+        result.groupedOutput.task(':failing').output == 'hello'
+        result.output.contains(styled("> Task :failing", Ansi.Color.RED, Ansi.Attribute.INTENSITY_BOLD))
+    }
+
+    def "group header is printed red if task failed and there is no output"() {
```

Comment:
it would be good if we had this test coverage for each of the console modes, not just rich.

Reply:
Looks like we can just add a rich console subclass for `AbstractConsoleVerboseRenderingFunctionalTest`

Reply:
We should also perhaps rename that class, as it's covering the task header behaviour